src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -94,6 +107,23 @@ void ClusterTaskManager::ScheduleAndDispatchTasks() {
         RAY_LOG(DEBUG) << "No node found to schedule a task "
                        << task.GetTaskSpecification().TaskId() << " is infeasible?"
                        << is_infeasible;
+
+        if (task.GetTaskSpecification().IsNodeAffinitySchedulingStrategy() &&
```

Comment:
ah. should this logic be part of `cluster_resource_scheduler_->GetBestSchedulableNode` ? (it has an `is_infeasible` argument)

Reply:
Discussed offline: The cancel logic is in cluster_task_manager but we can propagate the infeasible reason from scheduling policy in the future.